File: signout_test.go
package gatewayauth

import "testing"

func TestSignOutURL(t *testing.T) {
	t.Parallel()

	cases := []struct {
		prefix, redirect, want string
	}{
		{"", "", "/oauth2/sign_out"},
		{"/oauth2", "", "/oauth2/sign_out"},
		{"_gwauth", "", "/_gwauth/sign_out"},
		{"/_gwauth/", "", "/_gwauth/sign_out"},
		{"/oauth2", "/", "/oauth2/sign_out?rd=%2F"},
	}

	for _, c := range cases {
		if got := SignOutURL(c.prefix, c.redirect); got != c.want {
			t.Errorf("SignOutURL(%q,%q) = %q, want %q", c.prefix, c.redirect, got, c.want)
		}
	}
}

File: ts/src/signout.test.ts
import { describe, expect, it } from "vitest";

import { signOutUrl } from "./signout.js";

describe("signOutUrl", () => {
  it("defaults the proxy prefix to /oauth2", () => {
    expect(signOutUrl()).toBe("/oauth2/sign_out");
  });

  it("honors a custom prefix and normalizes slashes", () => {
    expect(signOutUrl("_gwauth")).toBe("/_gwauth/sign_out");
    expect(signOutUrl("/_gwauth/")).toBe("/_gwauth/sign_out");
  });

  it("adds the redirect as rd", () => {
    expect(signOutUrl("/oauth2", "/")).toBe("/oauth2/sign_out?rd=%2F");
  });
});
